tony-core/src/test/java/com/linkedin/tony/TestEventHandler.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright 2018 LinkedIn Corporation. All rights reserved. Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.tony;
+
+import com.linkedin.tony.events.ApplicationInited;
+import com.linkedin.tony.events.Event;
+import com.linkedin.tony.events.EventHandler;
+import com.linkedin.tony.events.EventType;
+import com.linkedin.tony.util.HistoryFileUtils;
+import com.linkedin.tony.util.Utils;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import org.apache.avro.file.DataFileStream;
+import org.apache.avro.file.DataFileWriter;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.specific.SpecificDatumReader;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.HdfsConfiguration;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.mockito.Mockito.*;
+import static org.testng.Assert.*;
+
+
+public class TestEventHandler {
+  private static final Log LOG = LogFactory.getLog(TestEventHandler.class);
+  private FileSystem fs = null;
+  private BlockingQueue<Event> eventQueue;
+  private Thread eventHandlerThread;
+  private Event eEventWrapper;
+  private ApplicationInited eAppInitEvent = new ApplicationInited("app123", 1, "fakehost");
+  private Path jobDir = new Path("./src/test/resources/jobDir");
+  private TonyJobMetadata metadata = new TonyJobMetadata();
+
+  private List<Event> parseEvents(FileSystem fs, Path jobDir, TonyJobMetadata metadata) {
```

Comment:
Please don't duplicate the code in `ParserUtils.parseEvents()` here. You can move `ParserUtils` from the tony-history-server module to the tony-core module to avoid the cyclic dependency.